test-data/unit/check-typevar-tuple.test

```diff
@@ -346,4 +346,20 @@ expect_variadic_array(u)
 expect_variadic_array_2(u)
 
 
+[builtins fixtures/tuple.pyi]
+
+[case testPep646TypeVarStarArgs]
+from typing import Tuple
+from typing_extensions import TypeVarTuple, Unpack
+
+Ts = TypeVarTuple("Ts")
+
+# TODO: add less trivial tests with prefix/suffix etc.
+# TODO: add tests that call with a type var tuple instead of just args.
+def args_to_tuple(*args: Unpack[Ts]) -> Tuple[Unpack[Ts]]:
```

Comment:
Will this PR work for a non-typevartuple use case? I have seen people use something like this `def foo(*args: Unpack[tuple[int, str]])` (but it will be probably more useful with a type alias, to re-use in different functions, similar to how we allow TypedDicts to re-use types for `**kwds`).

Btw currently this use case may cause a crash, see first example in https://github.com/python/mypy/pull/13790#issuecomment-1266926539

If this PR helps with this direction, could you please add a test case? Otherwise consider this a sub-feature request.

Reply:
Another common use case  with `*args` which crashes currently.
```py
from typing import Callable
from typing_extensions import TypeVarTuple, Unpack

Ts = TypeVarTuple("Ts")

def a(var: int) -> None: ...

def func(f: Callable[[Unpack[Ts]], None], *args: Unpack[Ts]) -> None:
    f(*args)

func(a, 42)
```